File: android/core/models/src/main/kotlin/com/nexussoft/verbum/models/BibleEntity.kt
package com.nexussoft.verbum.models

/** docs/PRODUCT.md §22.1. `wireValue` is the string contract shared with iOS and the backend. */
enum class BibleEntityType(val wireValue: String) {
    PERSON("person"),
    PLACE("place"),
    EVENT("event"),
    THEME("theme"),
    PASSAGE("passage"),
    BOOK("book"),
    PROPHECY("prophecy"),
    ORIGINAL_TERM("originalTerm"),
    HISTORICAL_PERIOD("historicalPeriod");

    companion object {
        fun fromWireValue(value: String): BibleEntityType? = entries.firstOrNull { it.wireValue == value }
    }
}

/** docs/PRODUCT.md §22.1. */
data class BibleEntity(
    val id: String,
    val type: BibleEntityType,
    val name: String,
    val summary: String?,
)

File: VerbumKit/Sources/Models/BibleEntity.swift
/// Spec §22.1.
public enum BibleEntityType: String, Codable, Sendable, CaseIterable {
    case person
    case place
    case event
    case theme
    case passage
    case book
    case prophecy
    case originalTerm
    case historicalPeriod
}

/// Spec §22.1.
public struct BibleEntity: Identifiable, Codable, Equatable, Hashable, Sendable {
    public let id: String
    public let type: BibleEntityType
    public let name: String
    public let summary: String?

    public init(id: String, type: BibleEntityType, name: String, summary: String?) {
        self.id = id
        self.type = type
        self.name = name
        self.summary = summary
    }
}
